common/src/main/java/io/netty/util/ThreadDeathWatcher.java

```diff
@@ -121,10 +125,10 @@ public static boolean awaitInactivity(long timeout, TimeUnit unit) throws Interr
             throw new NullPointerException("unit");
         }
 
-        Thread watcherThread = ThreadDeathWatcher.watcherThread;
-        if (watcherThread != null) {
-            watcherThread.join(unit.toMillis(timeout));
-            return !watcherThread.isAlive();
+        Thread t = watcherThread.get();
+        if (t != null) {
```

Comment:
@Scottmitch I don't think so... This would be a breaking change. Maybe we should just deprecate this method and add a new one that does this. WDYT ?